minvws/nl-kat-coordination
# Create boefje variant

**NOTE:** This PR is a follow-up of PR #3400

------------------------

### Changes
This PR makes it possible to add a new Boefje, based on another Boefje.

Features added:
- "Add variant" button on Boefje detail page
- "Setup Boefje variant" form page that is pre-filled
   - Changes in the helper texts
   - Changed "Input object type" single select to multi select
   - This form page redirects to original Boefje detail page
- Boefje detail page
   - Variant should be added in the table
   - Confirmation should be visible
- Backend: add filter (oci_image) to get_plugins query, so we can collect all variants of one specific oci_image

### Issue link
Closes #3403 

### Demo
![afbeelding](https://github.com/user-attachments/assets/04881cb4-2315-45d4-80df-99501e43e4ac)
![afbeelding](https://github.com/user-attachments/assets/2e97225d-e63e-466d-8497-38c15c7524f7)


### QA notes
- Go to "Nmap TCP" boefje
- Check the "Variants" table. There should be one item (the current Boefje)
- Click "Add variant"
- Check if all the input fields are filled in (except for "name" and "description")
- Type a different name, make some small changes to the other input fields
- Click "Create variant"
- You should return to the "Nmap TCP" boefje
   - In the variants table, a new variant should be visible
   - There should be a confirmation message that the new variant has been created successfully
- Click on the name of the new variant and check the Boefje details page

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
